internals-js/src/values.ts

```diff
@@ -162,7 +162,12 @@ function objectEquals(a: {[key: string]: any}, b: {[key: string]: any}): boolean
     // Beware of false-negative due to getting undefined because the property is not
     // in args2.
     if (v2 === undefined) {
-      return v1 === undefined && b.hasOwnProperty(key);
+      // if objects were created with Object.create(null), they will not have
+      // a hasOwnProperty function, which makes it impossible to tell if
+      // the property is there explicitly or not
```

Comment:
Nit: it would be nice to slightly amend the comment. I think if we get to it in 6 months, it may be slightly confusing as phrased since `hasOwnProperty` is not mentioned anymore.